pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -29,188 +33,339 @@ type DockerRegistryServiceControllerOptions struct {
 	RegistryNamespace   string
 	RegistryServiceName string
 
-	DefaultDockerURL string
-
 	DockercfgController *DockercfgController
+
+	// DockerURLsIntialized is used to send a signal to this controller that it has the correct set of docker urls
+	DockerURLsIntialized chan struct{}
 }
 
 // NewDockerRegistryServiceController returns a new *DockerRegistryServiceController.
 func NewDockerRegistryServiceController(cl client.Interface, options DockerRegistryServiceControllerOptions) *DockerRegistryServiceController {
 	e := &DockerRegistryServiceController{
-		client: cl,
+		client:                cl,
+		dockercfgController:   options.DockercfgController,
+		registryLocationQueue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		secretsToUpdate:       workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		serviceName:           options.RegistryServiceName,
+		serviceNamespace:      options.RegistryNamespace,
+		dockerURLsIntialized:  options.DockerURLsIntialized,
 	}
 
-	_, e.serviceController = framework.NewInformer(
+	e.serviceCache, e.serviceController = framework.NewInformer(
 		&cache.ListWatch{
-			ListFunc: func(opts api.ListOptions) (runtime.Object, error) {
+			ListFunc: func(opts kapi.ListOptions) (runtime.Object, error) {
+				opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", options.RegistryServiceName)
```

Comment:
> Does this mean that we are going to cache only the registry service?

I'd like to get the new structure in, then port to the `SharedInformer` and we can do the comparison in the `enqueue` function.